test/utility/misc.c

```diff
@@ -138,6 +138,44 @@ testutil_clean_work_dir(const char *dir)
     free(buf);
 }
 
+/*
+ * testutil_deduce_build_dir --
+ *     Deduce the build directory.
+ */
+void
+testutil_deduce_build_dir(TEST_OPTS *opts)
+{
+    struct stat stats;
+
+    char path[512], pwd[512], token_path[512], stat_path[512];
+    char *token;
+    size_t len;
+
+    token_path[0] = '\0';
+
+    if (getcwd(pwd, sizeof(pwd)) == NULL)
+        testutil_die(ENOENT, "No such directory");
+
+    testutil_check(__wt_snprintf(path, sizeof(path), "%s/%s", pwd, opts->argv0));
```

Comment:
I'm not sure this works when a full pathname is used for argv0.  For example, if the cur directory is `/foo/bar` and we run `/some/dir/build/bench/tiered/push_pull`, then path will be set to `/foo/bar/some/dir/build/bench/tiered/push_pull` (a directory that doesn't exist).  Will it find a `wt`?

I think the code needs to test `opts->argv0[0]`.  If it is `'/'`, then set `path` to it directly (no need for pwd).  If not `'/'`, then what you have is correct.

Reply:
Fixed